sdk/identity/identity/src/credentials/workloadIdentityCredential.ts

```diff
@@ -56,21 +62,66 @@ export class WorkloadIdentityCredential implements TokenCredential {
     const workloadIdentityCredentialOptions = options ?? {};
     const tenantId = workloadIdentityCredentialOptions.tenantId || process.env.AZURE_TENANT_ID;
     const clientId = workloadIdentityCredentialOptions.clientId || process.env.AZURE_CLIENT_ID;
+
     this.federatedTokenFilePath =
-      workloadIdentityCredentialOptions.tokenFilePath || process.env.AZURE_FEDERATED_TOKEN_FILE;
+      workloadIdentityCredentialOptions?.tokenFilePath || process.env.AZURE_FEDERATED_TOKEN_FILE;
+    this.serviceConnectionId = workloadIdentityCredentialOptions.serviceConnectionId;
+    if (!tenantId || !clientId) {
```

Comment:
nit: I would move this to line 65 to keep it next to the code that sets clientId and tenantId